ADC_DMA/Core/Src/main.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * 	Implement a routine that reads the internal voltage reference (ADC_IN17)
+ * 	and internal temperature sensor (ADC_IN18).
+ * 	The internal reference voltage must be in millivolts and the temperature in Celsius.
+ *
+ * 	1 - Polling		[X]
+ * 	2 - Interrupt	[X]
+ * 	3 - DMA			[/]
+ */
+
+#include "stm32l053xx.h"
+#include "macros.h"
+
+struct measure {
+	uint16_t	v_ref;			//internal reference voltage measured (after calculation)
+	int16_t		temperature;	//temperature in Celsius degrees measured (after calculation)
+}ADC_measure;	//struct with all the information read by the ADC channel
+
```

Comment:
Nice use of a struct! I would give you a few more tips:
declare de struct as a typedef like this:
```
typedef struct {
	uint16_t	v_ref;			//internal reference voltage measured (after calculation)
	int16_t		temperature;	//temperature in Celsius degrees measured (after calculation)
}sADC_measure;
```

Now you have a type called `sADC_measure` (s stands for struct) and you can initialise it like if it was an int. Put `sADC_measure ADC_measure`inside your main function and you will have a local ADC_measure var.

Reply:
I think you can also initialise it filled with zero values by doing `sADC_measure ADC_measure = { 0 }`

Reply:
Ok. I'll read a little about typedef right after

Reply:
did the initialisation `= {0}` worked?

Reply:
yep

Reply:
nice